src/main/java/frc/robot/subsystems/WheelSubsystem.java

```diff
@@ -3,62 +3,83 @@
 import com.revrobotics.CANSparkMax;
 import com.revrobotics.SparkMaxPIDController;
 import com.revrobotics.CANSparkMax.ControlType;
-import com.revrobotics.CANSparkMax.IdleMode;
-import com.revrobotics.CANSparkMaxLowLevel.MotorType;
 
 import edu.wpi.first.math.MathUtil;
 import edu.wpi.first.math.controller.PIDController;
 import edu.wpi.first.math.geometry.Rotation2d;
 import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
 import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
 import edu.wpi.first.wpilibj.AnalogEncoder;
-import edu.wpi.first.wpilibj.Encoder;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import frc.robot.Constants.DrivetrainSubsystem;
 
 public class WheelSubsystem extends SubsystemBase{
 
     
     private CANSparkMax angleMotor;
     private CANSparkMax speedMotor;
-    private PIDController pidController;
+    private PIDController anglePIDController;
+    private SparkMaxPIDController speedPIDController;
     private AnalogEncoder turningEncoder;    
     
     Translation2d location;
 
-    private double targetVoltage = 0;
+    private double targetSpeed = 0;
 
     private double angleSetpoint = 0;
+    private final NetworkTableInstance ntInstance = NetworkTableInstance.getDefault();
+    private final NetworkTable table = ntInstance.getTable("/components/drivetrain");
 
+    private final NetworkTableEntry speedTarget = table.getEntry("speedTarget");
+    private final NetworkTableEntry velocity = table.getEntry("wheelvelocity");
 
     public WheelSubsystem (CANSparkMax angleMotor, CANSparkMax speedMotor, AnalogEncoder turningEncoder, Translation2d location) {
         this.angleMotor = angleMotor;
         this.speedMotor = speedMotor;
-        this.location = location;
         this.turningEncoder = turningEncoder;
+        this.location = location;
+
+        this.speedMotor.getEncoder().setPosition(0);
+        this.speedMotor.getEncoder().setPositionConversionFactor(0.33/8.33);
+        this.speedMotor.getEncoder().setVelocityConversionFactor(this.speedMotor.getEncoder().getPositionConversionFactor() / 60.0);
+        this.speedPIDController = this.speedMotor.getPIDController();
+
+        speedPIDController.setP(0.00);
+        speedPIDController.setI(0.00);
+        speedPIDController.setD(0.00);
+        speedPIDController.setFF(0.259);
```

Comment:
so it just always runs forward?

Reply:
FeedForward was enough for accurate velocity, so we didn't bother with PID.